src/main/asciidoc/index.adoc

```diff
@@ -17,503 +17,801 @@
-```
-eyJhbGciOiJSUzI1NiIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..CV_KdWKFu3PyIAN_qv9R89tC0OQBLow7ycF_TpG7pwvZi4jx_v44gLPvZz814DURTZOTUWHGM09BYVwaJYTPGZWOkjRO8ynJy37NGzQ6Amu3o1vnFraPbKySgqCqNiNEbBAa8wCrbgMCww6NTT5tB2IKjWOMwdwVbnnb4JB9Q5CUKGWevrKnGhl-fGf96AU2YQGeBPlHo_BlpC--UOsgRdMrgHyl3t4eiQRw68CfLGokRQUIRorzM_Aq5bf6-H88eexcJ6W9PT7RzRtXhEl9GNf-0NGKMJep3mEaRraXG-t8fxDKj2nrxmEZZV4Riog5tpaj9NmL8veAkjweA7RdTA
-```
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+RESTful services MUST NOT expose internal-use or management endpoints to the public Internet.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+In case of failure, RESTful services MUST NOT pass technical details in error messages.
 
-* Prepare your HTTP message using the JWS.
-** The access tokens in OAuth 2.0 are of type *bearer*, meaning the client needs to pass the token with each request.
-** The followings are the expected HTTP headers:
+Instead, it is RECOMMENDED that a unique, randomly generated correlation identifier be provided in the response to
+facilitate future troubleshooting.
+====
 
-[width="100%",cols="22%,19%,17%,42%",]
-|===
-|*HTTP Header* |*Description* |*Required* |*Value*
-|Authorization |Must be set to the access token. |Must |Bearer \{Access Token}
-|BelGov-_Org_-JWS |JWS object to be sent. |Must |\{Detached JWS}
-|Content-Type |Content type of the payload. |Optional a|
-\{JSON body} -> application/jose+json
-
-\{Non JSON body} -> application/jose
-
-|===
-
-=== Using JWE for Encryption of Request and Response
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+RESTful services MUST include the “Content-Type” HTTP header in all responses.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+RESTful services MUST set the “Cache-Control” HTTP header to no-store to prevent caching of sensitive information.
+====
+
+Cross-Origin Resource Sharing (CORS) is an HTTP header-based mechanism used to allow or disallow resource sharing
+between different origins (i.e., domains).
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+RESTful services SHOULD allow only a specific domain or set of domains.
+
+Non-public RESTful services MUST NOT allow all domains (origins).
+====
+
+== HTTP message body
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+All requests to RESTful services handling Category 1-2 data SHOULD be digitally signed by the API consumer using JWS.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+All responses from RESTful services handling Category 1-2 data MUST be digitally signed by the API provider using JWS.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+To avoid obfuscating HTTP body data by wrapping the payload in a JWS envelope, the HTTP body MUST be represented as a
+detached JWS structure.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The signature MUST be provided in a custom “x-jws-signature” HTTP header.
```

Comment:
Don't know if it's the case here, but if the x- header is already (pseudo-)standard, the prefix is still allowed.

Has the HTTP message signatures IETF draft been considered?

https://httpwg.org/http-extensions/draft-ietf-httpbis-message-signatures.html#name-request-response-signature-

Reply:
Yes, I've thought about it and I've read quite a few Amazon publications on the subject. My conclusion is that, while it is very promising (is it even fair to say promising considering that it has been used in production by Amazon for a few years now?), since it is not standardized, it has spotty support, which would make it difficult to implement.

Reply:
nothing to change.